Lesley week 1 first

LslyKng opened the pull request.

djgrant commented on week-2/D-predicates/exercise.js:
```diff
@@ -2,16 +2,19 @@
 // Finish the predicate function to test if the passed number is between 0 and 10
-function isBetweenZeroAnd10(number) {
+  function isBetweenZeroAnd10(number) {
+  var number = 0
```
This function should `return` either `true` or `false` depending on whether `number` is between zero and 10 or not.

You can use an if statement to check if `number` passes the condition. 

``` 
if (condition) { 
  return true;
} else {
  return false;
}
```

What should the condition be?

There are actually two conditions: 
- bigger than or equal to zero
- less than or equal to 10

Both those conditions should be true. What logical operator can you use for that?

djgrant commented on week-2/E-conditionals/exercise.js:
```diff
@@ -5,14 +5,15 @@
 var role2 = "student";
-
-
+ {
```
Don't need the braces (`{` `}`) here.

djgrant commented on week-2/J-array-get-set/exercise.js:
```diff
@@ -5,11 +5,11 @@
 
 function last(arr) {
-  return; // complete this statement
+  return arr [2] ; // complete this statement
```
What if the parameter `arr` passed to this function has 4 items? Would this still work?

Try to re-write this without hardcoding the last value.

LslyKng commented:
Thanks

On Thu, 1 Mar 2018, 06:58 Daniel Grant, <notifications@github.com> wrote:

> *@djgrant* approved this pull request.
>
> Great start, @LslyKng <https://github.com/lslykng>!
>
> Looking forward to seeing more exercises. By the way, don't worry about
> having everything perfect before pushing your work. If there's something
> you don't get yet, I, or another mentor, can take a look and give you
> advice in the PR.
> ------------------------------
>
> In week-2/D-predicates/exercise.js
> <https://github.com/CodeYourFuture/js-exercises/pull/12#discussion_r171475421>
> :
>
> >  }
>
>  // Finish the predicate function to test if the passed number is between 0 and 10
> -function isBetweenZeroAnd10(number) {
> +  function isBetweenZeroAnd10(number) {
> +  var number = 0
>
> This function should return either true or false depending on whether
> number is between zero and 10 or not.
>
> You can use an if statement to check if number passes the condition.
>
> if (condition) {
>   return true;
> } else {
>   return false;
> }
>
> What should the condition be?
>
> There are actually two conditions:
>
>    - bigger than or equal to zero
>    - less than or equal to 10
>
> Both those conditions should be true. What logical operator can you use
> for that?
> ------------------------------
>
> In week-2/E-conditionals/exercise.js
> <https://github.com/CodeYourFuture/js-exercises/pull/12#discussion_r171475503>
> :
>
> >  var name = "Daniel";
>  var role = "mentor";
>
>  var name2 = "Harun";
>  var role2 = "student";
> -
> -
> + {
>
> Don't need the braces ({ }) here.
> ------------------------------
>
> In week-2/J-array-get-set/exercise.js
> <https://github.com/CodeYourFuture/js-exercises/pull/12#discussion_r171475682>
> :
>
> >  }
>
>  function last(arr) {
> -  return; // complete this statement
> +  return arr [2] ; // complete this statement
>
> What if the parameter arr passed to this function has 4 items? Would this
> still work?
>
> Try to re-write this without hardcoding the last value.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/CodeYourFuture/js-exercises/pull/12#pullrequestreview-100330867>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/Ah28IKgVRmCmqVLsGNJbKzjLTGoGyIkkks5tZ5v8gaJpZM4SR26G>
> .
>

tekul commented on week-2-practise/level-1/5-indexOf.js:
```diff
@@ -3,12 +3,12 @@
   ---------------
   3
-*/
+*
```
Does this work without any errors if you run it? You've accidentally deleted the `/` from the comment end marker.

tekul commented on week-2-practise/level-1/6-map.js:
```diff
@@ -11,7 +11,7 @@ function tidyUpString(str) {
 
 var mentors = ["/Daniel ", "irina ", " Gordon", "ashleigh "];
-var mentorsTidy; // ONLY EDIT THIS LINE
+var mentorsTidy = str.map (tidyUpString); // ONLY EDIT THIS LINE
```
What's `str` here? You need to call `map` on the array and it will apply the function to each member.

tekul commented on week-2/D-predicates/exercise.js:
```diff
@@ -8,13 +8,13 @@
 // Finish the predicate function to test if the passed number is between 0 and 10
   function isBetweenZeroAnd10(number) {
-  var number = 0
+    var number = (5) 
```
This needs to check whether `number > 0` and also `number < 10`, so we need to combine two boolean conditions using the "and" operator (which is called `&&` in Javascript).

tekul commented on week-2/F-logical-operators/exercise.js:
```diff
@@ -11,6 +11,10 @@ var cssLevel = 4;
+  if (num>5) {return true}
+  return false 
+}
 var htmlLevelAbove5;
```
This exercise is similar to the above in that it wants you to combine booleans using the operators:

- and (`&&`) 
- or (`||`)

tekul commented on week-2/F-logical-operators/exercise2.js:
```diff
@@ -5,18 +5,34 @@
+  else {return false}
+}
+
+//function isBetween5and10(number)
```
You've commented out the function names here? `//` is a line comment marker.

Also, remember what we said on Saturday about what to do if you find yourself writing:

```
if (something) {
    return true;
} else {
    return false;
}
```
:-) ?

tekul commented on week-2/M-array-map/exercise.js:
```diff
@@ -10,8 +10,8 @@ function double(num) {
 var numbers = [1, 2, 3, 4];
-var numbersDoubled; // complete this statement (use map and the double function)
-
+var numbersDoubled = numbers.map (double); // complete this statement (use map and the double function)
```
Looks good :)

tekul commented on week-2/L-array-methods-2/exercise2.js:
```diff
@@ -7,7 +7,11 @@
 
-function capitalise(str) {}
+function capitalise(str) {
+  var str = split (separator) 'h');
```
I'm a bit confused here. What's this doing? It's quite a tricky problem. A good idea is to search for the functions it suggests to use. I just did and I found [this page](https://www.w3schools.com/jsref/jsref_split.asp), which also has a useful tip:

> Tip: If an empty string ("") is used as the separator, the string is split between each character.

Playing with that and the `join` function in node:

```
> "daniel".split("");
[ 'd', 'a', 'n', 'i', 'e', 'l' ]
> "daniel".split("").join();
'd,a,n,i,e,l'
> "daniel".split("").join("");
'daniel'
```
So if you can find out how to convert a string to upper case, you can split the string into an array  of characters, upper case the first element of the array, then join them back together.

LslyKng commented:
Thanks Luke I'll have a look tomorrow

On Wed, 7 Mar 2018, 19:31 ___, <notifications@github.com> wrote:

> *@tekul* commented on this pull request.
>
> Excellent work Lesley. Check my comments and let me know if you have any
> questions. And keep the exercises coming 🥇 💯
> ------------------------------
>
> In week-1-practise/level-1/4-logic-error-1.js
> <https://github.com/CodeYourFuture/js-exercises/pull/12#discussion_r172944241>
> :
>
> > @@ -1,10 +1,10 @@
>  // The syntax for this function is valid but it has an error, find it and fix it.
>
>  function trimWord(word) {
> -  return wordtrim();
> +  return trimWord();
>
> This function is calling itself with no parameters. What happens if you
> run it? Hint: There's a standard trim() method which you can call on a
> string.
> ------------------------------
>
> In week-1-practise/level-2/4-currency-conversion-2.js
> <https://github.com/CodeYourFuture/js-exercises/pull/12#discussion_r172945451>
> :
>
> > @@ -7,8 +7,11 @@
>    Find a way to add 1% to all currency conversions (think about the DRY principle)
>  */
>
> -function convertToUSD(price) {}
> -function convertToBRL(price) {}
> +function convertToUSD(price) {
> +}
> +function convertToBRL(price) {
> +
> +}
>
> The question doesn't tell you the exchange rates, so you can use the
> previous one dollars. When I was doing this with Harun we just looked up
> the real
> <https://www.google.co.uk/search?sclient=psy-ab&site=&source=hp&q=1%20gbp%20in%20brl>
> which gives 4.5. So it's kind of the same as a exercise before, but you
> need to add the 1% surcharge in both cases.
> ------------------------------
>
> In week-1-practise/level-3/1-piping.js
> <https://github.com/CodeYourFuture/js-exercises/pull/12#discussion_r172947586>
> :
>
> > @@ -10,7 +10,7 @@
>    As a learning exercise (you wouldn't do this normally), do all of the above in 1 line of code
>  */
>
> -var startingValue = 5;
> +var startingValue = (5*2)+10; return result = startingValue
>
> Remember that you can only use return inside a function. This wants you
> create a variable called result then use the functions below to do what
> it asks - i.e. call multiply, add and format.
> ------------------------------
>
> In week-1/C-hello-world/exercise.js
> <https://github.com/CodeYourFuture/js-exercises/pull/12#discussion_r172947993>
> :
>
> > @@ -1 +1,11 @@
> -console.log("Hello world I just started learning JS");
> +var greeting='java script woohoo!'
> +var greeting2='so much fun'
> +var greeting3='cannot wait to do more'
> +console.log (greeting,greeting2,greeting3)
> +
> +var name='Lesley'
> +var nameStart='This is true says '
> +console.log (nameStart,name)
> +
> +var question='Do you believe me?'
> +console.log (question)
>
> Hmm. I'm not sure :-). Should I?
> ------------------------------
>
> In week-1/I-floats/exercise.js
> <https://github.com/CodeYourFuture/js-exercises/pull/12#discussion_r172948601>
> :
>
> > \ No newline at end of file
> +var numberOfMentors = 8
> +var sum = (numberOfStudents + numberOfMentors)
> +var percent = Math.round (numberOfStudents/sum*100)
> +var percent2 = Math.round (numberOfMentors/sum*100)
>
> This looks pretty good. Perhaps just give these more meaningful names,
> e.g. percentageOfStudents ? And don't forget your semicolons at the end
> of lines.
> ------------------------------
>
> In week-2-practise/level-1/5-indexOf.js
> <https://github.com/CodeYourFuture/js-exercises/pull/12#discussion_r172949822>
> :
>
> >
>  console.log(index);
>
>  /*
>    EXPECTED RESULT
>    ---------------
>    3
> -*/
> +*
>
> Does this work without any errors if you run it? You've accidentally
> deleted the / from the comment end marker.
> ------------------------------
>
> In week-2-practise/level-1/6-map.js
> <https://github.com/CodeYourFuture/js-exercises/pull/12#discussion_r172950254>
> :
>
> > @@ -11,7 +11,7 @@ function tidyUpString(str) {
>  }
>
>  var mentors = ["/Daniel ", "irina ", " Gordon", "ashleigh "];
> -var mentorsTidy; // ONLY EDIT THIS LINE
> +var mentorsTidy = str.map (tidyUpString); // ONLY EDIT THIS LINE
>
> What's str here? You need to call map on the array and it will apply the
> function to each member.
> ------------------------------
>
> In week-2/D-predicates/exercise.js
> <https://github.com/CodeYourFuture/js-exercises/pull/12#discussion_r172951020>
> :
>
> >      return number<0;
>  }
>
>  // Finish the predicate function to test if the passed number is between 0 and 10
>    function isBetweenZeroAnd10(number) {
> -  var number = 0
> +    var number = (5)
>
> This needs to check whether number > 0 and also number < 10, so we need
> to combine two boolean conditions using the "and" operator (which is called
> && in Javascript).
> ------------------------------
>
> In week-2/F-logical-operators/exercise.js
> <https://github.com/CodeYourFuture/js-exercises/pull/12#discussion_r172951599>
> :
>
> > @@ -11,6 +11,10 @@ var cssLevel = 4;
>
>  // Finish the statement to check whether HTML, CSS knowledge are above 5
>  // (hint: use the comparison operator from before)
> +functions satisfiesRequirements(num) {
> +  if (num>5) {return true}
> +  return false
> +}
>  var htmlLevelAbove5;
>
> This exercise is similar to the above in that it wants you to combine
> booleans using the operators:
>
>    - and (&&)
>    - or (||)
>
> ------------------------------
>
> In week-2/F-logical-operators/exercise2.js
> <https://github.com/CodeYourFuture/js-exercises/pull/12#discussion_r172952379>
> :
>
> > @@ -5,18 +5,34 @@
>    Update the code so that you get the expected result.
>  */
>
> -function isNegative() {}
> +function isNegative(number) {
> +  if (number<0) {return true}
> +  else {return false}
> +}
> +
> +//function isBetween5and10(number)
>
> You've commented out the function names here? // is a line comment marker.
>
> Also, remember what we said on Saturday about what to do if you find
> yourself writing:
>
> if (something) {
>     return true;
> } else {
>     return false;
> }
>
> :-) ?
> ------------------------------
>
> In week-2/M-array-map/exercise.js
> <https://github.com/CodeYourFuture/js-exercises/pull/12#discussion_r172952883>
> :
>
> > @@ -10,8 +10,8 @@ function double(num) {
>  }
>
>  var numbers = [1, 2, 3, 4];
> -var numbersDoubled; // complete this statement (use map and the double function)
> -
> +var numbersDoubled = numbers.map (double); // complete this statement (use map and the double function)
>
> Looks good :)
> ------------------------------
>
> In week-2/L-array-methods-2/exercise2.js
> <https://github.com/CodeYourFuture/js-exercises/pull/12#discussion_r172954982>
> :
>
> > @@ -7,7 +7,11 @@
>    Tip: use the string method .split() and the array method .join()
>  */
>
> -function capitalise(str) {}
> +function capitalise(str) {
> +  var str = split (separator) 'h');
>
> I'm a bit confused here. What's this doing? It's quite a tricky problem. A
> good idea is to search for the functions it suggests to use. I just did and
> I found this page <https://www.w3schools.com/jsref/jsref_split.asp>,
> which also has a useful tip:
>
> Tip: If an empty string ("") is used as the separator, the string is split
> between each character.
>
> Playing with that and the join function in node:
>
> > "daniel".split("");
> [ 'd', 'a', 'n', 'i', 'e', 'l' ]
> > "daniel".split("").join();
> 'd,a,n,i,e,l'
> > "daniel".split("").join("");
> 'daniel'
>
> So if you can find out how to convert a string to upper case, you can
> split the string into an array of characters, upper case the first element
> of the array, then join them back together.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/CodeYourFuture/js-exercises/pull/12#pullrequestreview-102049263>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/Ah28ICL_mtDJdYAgQiapv5AD25fQxcRbks5tcDMqgaJpZM4SR26G>
> .
>

LslyKng commented:
In week-2/J-array-get-set/exercise.js
<https://github.com/CodeYourFuture/js-exercises/pull/12#discussion_r171475682>
:


i stole this from internet and it works but I don't know why


function first(arr) {
return arr[0]; // complete this statement
}

function last(arr) {
return arr.slice(-1)[0]; // complete this statement
}

>  }

 function last(arr) {
-  return; // complete this statement
+  return arr [2] ; // complete this statement

What if the parameter arr passed to this function has 4 items? Would this
still work?

Try to re-write this without hardcoding the last value.

----------------------------------------------------------------------------------------------------

Both those conditions should be true. What logical operator can you use for
that?

condition && condition

On 1 March 2018 at 06:58, Daniel Grant <notifications@github.com> wrote:

> *@djgrant* approved this pull request.
>
> Great start, @LslyKng <https://github.com/lslykng>!
>
> Looking forward to seeing more exercises. By the way, don't worry about
> having everything perfect before pushing your work. If there's something
> you don't get yet, I, or another mentor, can take a look and give you
> advice in the PR.
> ------------------------------
>
> In week-2/D-predicates/exercise.js
> <https://github.com/CodeYourFuture/js-exercises/pull/12#discussion_r171475421>
> :
>
> >  }
>
>  // Finish the predicate function to test if the passed number is between 0 and 10
> -function isBetweenZeroAnd10(number) {
> +  function isBetweenZeroAnd10(number) {
> +  var number = 0
>
> This function should return either true or false depending on whether
> number is between zero and 10 or not.
>
> You can use an if statement to check if number passes the condition.
>
> if (condition) {
>   return true;
> } else {
>   return false;
> }
>
> What should the condition be?
>
> There are actually two conditions:
>
>    - bigger than or equal to zero
>    - less than or equal to 10
>
> Both those conditions should be true. What logical operator can you use
> for that?
> ------------------------------
>
> In week-2/E-conditionals/exercise.js
> <https://github.com/CodeYourFuture/js-exercises/pull/12#discussion_r171475503>
> :
>
> >  var name = "Daniel";
>  var role = "mentor";
>
>  var name2 = "Harun";
>  var role2 = "student";
> -
> -
> + {
>
> Don't need the braces ({ }) here.
> ------------------------------
>
> In week-2/J-array-get-set/exercise.js
> <https://github.com/CodeYourFuture/js-exercises/pull/12#discussion_r171475682>
> :
>
> >  }
>
>  function last(arr) {
> -  return; // complete this statement
> +  return arr [2] ; // complete this statement
>
> What if the parameter arr passed to this function has 4 items? Would this
> still work?
>
> Try to re-write this without hardcoding the last value.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/CodeYourFuture/js-exercises/pull/12#pullrequestreview-100330867>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/Ah28IKgVRmCmqVLsGNJbKzjLTGoGyIkkks5tZ5v8gaJpZM4SR26G>
> .
>
